calour/tests/test_analysis.py

```diff
@@ -48,40 +48,52 @@ def test_diff_abundance(self):
         for cid in expected_ids:
             print(self.test1.feature_metadata.index[cid])
             self.assertIn(self.test1.feature_metadata.index[cid], dd.feature_metadata.index)
+
+    def test_correlation(self):
```

Comment:
split to functions
also there is testing with the complex dataset which gets different results for spearman vs. pearson